android/java/org/chromium/chrome/browser/undo_tab_close_snackbar/BraveUndoBarController.java

```diff
@@ -0,0 +1,40 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.undo_tab_close_snackbar;
+
+import android.content.Context;
+
+import androidx.annotation.Nullable;
+
+import org.chromium.base.BravePreferenceKeys;
+import org.chromium.base.supplier.Supplier;
+import org.chromium.chrome.browser.preferences.ChromeSharedPreferences;
+import org.chromium.chrome.browser.tab.Tab;
+import org.chromium.chrome.browser.tabmodel.TabModelSelector;
+import org.chromium.chrome.browser.ui.messages.snackbar.SnackbarManager.SnackbarManageable;
+
+import java.util.List;
+
+public class BraveUndoBarController extends UndoBarController {
+    public BraveUndoBarController(
+            Context context,
+            TabModelSelector selector,
+            SnackbarManageable snackbarManageable,
+            @Nullable Supplier<Boolean> dialogVisibilitySupplier) {
+        super(context, selector, snackbarManageable, dialogVisibilitySupplier);
+    }
+
+    @Override
+    protected void showUndoBar(List<Tab> closedTabs, boolean isAllTabs) {
+        boolean should_show_undo_bar =
+                ChromeSharedPreferences.getInstance()
+                        .readBoolean(BravePreferenceKeys.SHOW_UNDO_ON_TAB_CLOSED, true);
+        if (!should_show_undo_bar) {
+            return;
```

Comment:
Looking on the description and implementation of `showUndoBar` https://github.com/chromium/chromium/blob/128.0.6613.107/chrome/android/java/src/org/chromium/chrome/browser/undo_tab_close_snackbar/UndoBarController.java#L149 there are calls or undo bar dismissal or on unno button click. You have to commit tabs closure https://github.com/chromium/chromium/blob/128.0.6613.107/chrome/android/java/src/org/chromium/chrome/browser/undo_tab_close_snackbar/UndoBarController.java#L285.

Reply:
Ahh, Then Can I do it like this?
```suggestion
            if (closedTabs.isEmpty()) return;
            Object actionData = closedTabs.size() == 1 ? closedTabs.get(0).getId() : closedTabs;
            onDismissNoAction(actionData);
  
            return;
```

Reply:
yes, that should work